joomla/joomla-cms
# [4.4] Fix event target in menu.js

### Summary of Changes
The core menu.js dynamically adds `aria-hidden=true` attributes to `ul` tags on menu level 2:

<img width="1050" alt="Bildschirmfoto 2024-09-04 um 10 54 57" src="https://github.com/user-attachments/assets/7f6be488-8360-4b57-87c5-08fefe7c581c">

However, on focus and hover, these attributes aren't always set to "false" (which would be correct as the previously invisible dropdown now is possible) as the JS has a bug. It uses the `target` attribute of the event, which is not the element that the eventhandler was bound to (=the parent `<li>`) but the element that triggered the event, i.e. a nested a or span tag.

It should use currentTarget.

See msdn:
https://developer.mozilla.org/en-US/docs/Web/API/Event/currentTarget


### Testing Instructions
Create a test site with a multi-level menu that uses the "default" menu layout. Make sure that the menu.js is loaded and the aria-hidden attribute is appended. Now hover over the element.


### Actual result BEFORE applying this Pull Request
aria-hidden isn't updated reliably.


### Expected result AFTER applying this Pull Request
aria-hidden is updated reliably.


### Link to documentations
Please select:
- [x ] No documentation changes for docs.joomla.org needed

- [ x] No documentation changes for manual.joomla.org needed
